### PR TITLE
GRO-447 Configure detekt

### PR DESCRIPTION
**Pull request by bishoyngendy**
**I kept the following detekt rules:** 
- Empty Function Block
- Complex Method 
- Nested Method

**Some other subjective rules, I added numbers so they are realistic somehow without fully removing the rules**
- Magic Numbers: Enabled it and it can be ignored only by providing named arguments
- Too Many Functions: Made Limit 12
- Long Parameter List: Made Limit 6

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+
+tasks.detekt {
+    reports {
+        xml.required.set(true)
+        xml.outputLocation.set(file("$projectDir/detekt/reports/detekt-report.xml"))
+
+        html.required.set(true)
+        html.outputLocation.set(file("$projectDir/detekt/reports/detekt-report.html"))
+    }
+}
```
Comment (bulatgaleev) [inline build.gradle.kts:23]: Do we need both xml and html reports?

**Review by bulatgaleev — COMMENTED** [review]
Looks good but can we configure it in a separate gradle file and apply it to the project one?

File: build.gradle.kts (hunk 1 above)
Comment (bishoyngendy) [reply build.gradle.kts:23]: @bulatgaleev I have tried to separate the config to the **utils.gradle.kts** file but I am receiving unresolved reference errors, I searched online but with no success, Are you familiar with this?

Comment (bulatgaleev) [reply build.gradle.kts:23]: I would keep xml report and have it as a part of our codebase

**Review by bulatgaleev — APPROVED** [review]

**Pull request merged**